consensus/src/data_manager.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::quorum_store::batch_reader::BatchReader;
+use aptos_crypto::HashValue;
+use aptos_logger::{debug, warn};
+use aptos_types::transaction::SignedTransaction;
+use arc_swap::ArcSwapOption;
+use consensus_types::common::DataStatus;
+use consensus_types::{
+    block::Block,
+    common::Payload,
+    proof_of_store::{LogicalTime, ProofOfStore},
+    request_response::WrapperCommand,
+};
+use executor_types::Error::BlockNotFound;
+use executor_types::*;
+use futures::channel::mpsc::Sender;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tokio::sync::oneshot;
+
+/// Responsible to extract the transactions out of the payload and notify QuorumStore about commits.
+/// If QuorumStore is enabled, has to ask BatchReader for the transaction behind the proofs of availability in the payload.
+pub struct PayloadManager {
+    quorum_store_enabled: AtomicBool,
+    data_reader: ArcSwapOption<BatchReader>,
+    quorum_store_wrapper_tx: ArcSwapOption<Sender<WrapperCommand>>,
```

Comment:
also I'm not a fan of multiple separate but related fields, maybe consider using something like this
```
enum PayloadManager {
  Direct,
  QuorumStore {
    data_reader: ..,
    request_tx: ..,
  }
}
```

Reply:
I don't understand how this can be an enum as It is long-lived and shared across threads.

Reply:
Specifically, can I have a method to change the enum itself? 
Or how would we switch to  QuorumStore on the epoch that enables it?

Reply:
@sasha8 @zekun000 looking at this some more, I think a fresh payload_manager should be created each epoch (and can be created as an enum as suggested above). After the changes made to payload_manager (fka data_manager) during this PR, there is no more shared state between epochs at all. Other than state_computer, it interacts only with components within the epoch (QSWrapper, BatchReader, BlockStore). We can change state_computer's new_epoch to swap out the newly created payload_manager.

Reply:
This makes things much cleaner. All QS components have a lifetime of an epoch.